### PR TITLE
LINK-1992 | add organization filter to event and registration search page

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Allow user to filter events and registrations by publisher organization. Support the feature in own events, event search and registrations pages.

## Closes
[LINK-1992](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1992)

## Screenshots
Screenshot 2024-05-17 at 11 44 52

Screenshot 2024-05-17 at 11 45 34

Screenshot 2024-05-17 at 11 46 01

[LINK-1992]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1992?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr393-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr393-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr393-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr393-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr393-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr393-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :laughing::tada::tada::tada:

File: src/domain/eventSearch/searchPanel/SearchPanel.tsx [hunk 1]
```diff
@@ -72,6 +81,12 @@ const SearchPanel: React.FC = () => {
     });
   };
 
+  const handleChangePublishers = (newPublishers: OptionType[]) => {
```
Comment (harriris-vincit) [inline src/domain/eventSearch/searchPanel/SearchPanel.tsx:84]: This function is duplicated three times. Maybe there could be one common utility function that takes as an additional argument the `setSearchState` function?

File: src/domain/events/searchPanel/__tests__/SearchPanel.test.tsx [hunk 2]
```diff
@@ -55,11 +65,21 @@ test('should search events with correct search params', async () => {
   const eventTypeCheckbox = screen.getByLabelText(/tapahtuma/i);
   await user.click(eventTypeCheckbox);
 
+  // Publisher filtering
```
Comment (harriris-vincit) [inline src/domain/events/searchPanel/__tests__/SearchPanel.test.tsx:68]: SonarCloud is detecting duplication here. Considering this whole test case, could there be a single common test utility function that does the test for both `src/domain/registrations/searchPanel/__tests__/SearchPanel.test.tsx` and `src/domain/events/searchPanel/__tests__/SearchPanel.test.tsx`?

File: src/domain/eventSearch/searchPanel/SearchPanel.tsx (hunk 1 above)
Comment (jorilindell) [reply src/domain/eventSearch/searchPanel/SearchPanel.tsx]: I implemented an useEventSearchHelpers hook which takes setSearchState as a parameter and return different type helper functions to set new search values.

File: src/domain/events/searchPanel/__tests__/SearchPanel.test.tsx (hunk 2 above)
Comment (jorilindell) [reply src/domain/events/searchPanel/__tests__/SearchPanel.test.tsx]: I added a separate shouldFilterEventsOrRegistrations function to test common search filters for events and registrations

**Comment by terovirtanen** [comment]
**LINKEDCOMPONENTS-UI branch is deployed to platta: https://linkedcomponents-pr393-ui.dev.hel.ninja** :rocket::rocket::rocket:

**Comment by terovirtanen** [comment]
## e2e tests result is __failed__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :crying_cat_face::anger::boom::boom:

**Comment by terovirtanen** [comment]
## e2e tests result is __success__ for https://linkedcomponents-pr393-ui.dev.hel.ninja :laughing::tada::tada::tada:

**Pull request merged**